datashader/core.py

```diff
@@ -399,8 +399,8 @@ def raster(self,
         if np.isclose(width_ratio, 0) or np.isclose(height_ratio, 0):
             raise ValueError('Canvas x_range or y_range values do not match closely enough with the data source to be able to accurately rasterize. Please provide ranges that are more accurate.')
 
-        w = max(int(np.ceil(self.plot_width * width_ratio)), 1)
-        h = max(int(np.ceil(self.plot_height * height_ratio)), 1)
```

Comment:
How did you establish that round was more accurate than ceil here?

Reply:
Will have to add some more tests, I found out because of shape mismatch errors when the padding was applied.